foundry.toml

```diff
@@ -36,6 +35,11 @@
   fail_on_revert = true
   runs = 20
 
+# Run only the code inside benchmark directory
+[profile.benchmark]
+  src = "benchmark"
```

Comment:
we specifically need `src = "benchmark"` - and we may need only this - without `test = "benchmark"`

otherwise, foundry will set `src = "src"` as the default and it would re-compile the source contracts - which we don't want

Reply:
I think I don't understand properly how it works. I ran the following tests with `bun run benchmark`:

1. `src = "benchmark"` and `test = "benchmark"` -> took 3m 55s. Re-run took the same time.
```bash
[⠊] Compiling...
[⠒] Compiling 139 files with Solc 0.8.23
[⠘] Solc 0.8.23 finished in 226.54s
Compiler run successful!
[⠊] Compiling...
[⠰] Compiling 114 files with Solc 0.8.23
[⠔] Solc 0.8.23 finished in 5.51s
Compiler run successful!
```
2. only `test = "benchmark"` -> took 3m 54s. Re-run took the same time.
```bash
[⠊] Compiling...
[⠢] Compiling 139 files with Solc 0.8.23
[⠆] Solc 0.8.23 finished in 226.13s
Compiler run successful!
[⠊] Compiling...
[⠒] Compiling 103 files with Solc 0.8.23
[⠘] Solc 0.8.23 finished in 4.81s
Compiler run successful!
```
3. only `src = "benchmark"` -> took 3m 53s. Re-run took the same time.
```bash
[⠊] Compiling...
[⠒] Compiling 139 files with Solc 0.8.23
[⠑] Solc 0.8.23 finished in 220.57s
Compiler run successful!
[⠊] Compiling...
[⠆] Compiling 114 files with Solc 0.8.23
[⠰] Solc 0.8.23 finished in 5.42s
Compiler run successful!
```

They all took the same time. Do you have any explanation? 🤔

Reply:
i will try to explain step by step

maybe it will be better to run the forge/foundry commands instead to understand better, also using `bun` will re-compile the contract always if we use optimized profile

```
# foundry.toml
```

Reply:
hmm, i get it now, it seems that:
```
# foundry.toml
[profile.benchmark]
  src = "benchmark"
  test = "benchmark"
```

and: 

```
# foundry.toml
[profile.benchmark]
  test = "benchmark"
```

to be the same, i am not sure why

Reply:
Yeah. I have asked this in the foundry telegram grp: https://t.me/foundry_support/53272. Lets see what they have to say. Until then, lets just use `test`.